README.md

```diff
@@ -1,247 +1,220 @@
-            iterationTimeUnit = "ms" // time unit for iterationTime, default is seconds
-        }   
+4.  **Using AllOpen Plugin**: The `allopen` plugin is used to satisfy JMH requirements, or all benchmark classes and methods should be `open`.
+
+    ```
+    allOpen {
+        annotation("org.openjdk.jmh.annotations.State")
     }
-    
-    // Setup targets
-    targets {
-        // This one matches compilation base name, e.g. 'jvm', 'jvmTest', etc
-        register("jvm") {
-            jmhVersion = "1.21" // available only for JVM compilations & Java source sets
-        }
-        register("js") {
-            // Note, that benchmarks.js uses a different approach of minTime & maxTime and run benchmarks
-            // until results are stable. We estimate minTime as iterationTime and maxTime as iterationTime*iterations
-            //
-            // You can configure benchmark executor - benchmarkJs or buildIn (works only for JsIr backend) with the next line:
-            // jsBenchmarksExecutor = JsBenchmarksExecutor.BuiltIn
+    ```
+
+5.  **Specifying Repository**: Ensure you have `mavenCentral()` in the list of repositories:
+
+        ```groovy
+        repositories {
+            mavenCentral()
         }
-        register("native")
-        register("wasm") // Experimental
-    }
-}
-```  
-  
-# Separate source sets for benchmarks
+        ```
+
+    </details>
+
+### Target-specific configurations
 
-Often you want to have benchmarks in the same project, but separated from main code, much like tests. Here is how:
+#### Kotlin/JS
 
-Define source set:
-```groovy
-sourceSets {
-    benchmarks
+For Kotlin/JS, include the `nodejs()` method call in the `kotlin` block:
+
+```kotlin
+kotlin {
+    js {
+        nodejs()
+    }
 }
 ```
 
-Propagate dependencies and output from `main` sourceSet. 
+For Kotlin/JS, both Legacy and IR backends are supported. However, simultaneous target declarations such as `kotlin.js.compiler=both` or `js(BOTH)` are not feasible. Ensure each backend is separately declared. For a detailed configuration example, please refer to the [build script of the kotlin-multiplatform example project](https://github.com/Kotlin/kotlinx-benchmark/blob/master/examples/kotlin-multiplatform/build.gradle).
+
+#### Multiplatform
+
+For multiplatform projects, add the `kotlinx-benchmark-runtime` dependency to the `commonMain` source set:
 
-```groovy
-dependencies {
-    benchmarksCompile sourceSets.main.output + sourceSets.main.runtimeClasspath 
+```kotlin
+kotlin {
+    sourceSets {
+        commonMain {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.4.8")
+            }
+        }
+    }
 }
 ```
 
-You can also add output and compileClasspath from `sourceSets.test` in the same way if you want 
-to reuse some of the test infrastructure.
+The platform-specific artifacts will be resolved automatically.
 
+### Benchmark Configuration
```

Comment:
It is possible to use kotlinx-benchmark in Java (`plugin { id 'java' }`) or Kotlin/JVM (`plugin { id 'org.jetbrains.kotlin.jvm' }`) projects. In this case you should register the source set containing your benchmarks as the target.
See `./example/java` and `./example/kotlin` example projects. See `JvmJavaTasks.kt` file to find out what our plugin does in such projects.

The library can also be used in kotlin multiplatform (`plugins { id 'org.jetbrains.kotlin.multiplatform' }`) projects. In this case you should register all targets you want to run benchmarks from. Benchmarks in the main compilation of the registered targets will be run. Another option is to register the compilation you want to run benchmarks from. See `./example/kotlin-multiplatform` sample project. See `BenchmarksExtension.kt` file to find the implementation of this logic.